Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -203,12 +223,52 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::push)
 // 23.1.3.31 Array.prototype.unshift ( ...items ), https://tc39.es/ecma262/#sec-array.prototype.unshift
 JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::unshift)
 {
-    auto* array = Array::typed_this(vm, global_object);
-    if (!array)
+    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    if (!this_object)
+        return {};
+    auto length = length_of_array_like(global_object, *this_object);
+    if (vm.exception())
+        return {};
+    auto arg_count = vm.argument_count();
+    size_t new_length = length + arg_count;
+    if (arg_count > 0) {
+
+        if (new_length > MAX_ARRAY_LIKE_INDEX) {
+            vm.throw_exception<TypeError>(global_object, ErrorType::ArrayMaxSize);
+            return {};
+        }
+        for (size_t k = length; k > 0; --k) {
+            auto from = k - 1;
+            auto to = k + arg_count - 1;
+
+            bool from_present = this_object->has_property(from);
+            if (vm.exception())
+                return {};
+            if (from_present) {
+                auto from_value = this_object->get(from);
```

Comment:
```suggestion
                auto from_value = this_object->get(from).value_or(js_undefined());
```

Reply:
For this and all the cases below is this needed?
We have checked that the object `has_property(from)` before so it should not be empty.
I have added it regardless.

Reply:
Yes, it's needed. If the this_object is a proxy targeting an array, it can lie about has_property and return an empty value for the get.

Reply:
Ah of course it is javascript after all. Wonder why test262 does not change but maybe it is because the other proxy parts do not work yet.